File: C++/6thOct-2.cpp
#include<iostream>
#include<vector>
using namespace std;

long long maxPrime = 100000000;
vector<bool> PrimeNumbers(maxPrime+1,true);

//Populate array with prime numbers takes n(log n) but here our n = maxPrime which is constant therefore takes constant time
void SieveOfEratosthenes()
{
    PrimeNumbers[0]=false;
    PrimeNumbers[1]=false;
    for(long long i=2; i*i<=maxPrime; i++) 
    { 
        if (PrimeNumbers[i]) 
        { 
            for (long long j=i*i; j<=maxPrime; j += i) 
                PrimeNumbers[j] = false; 
        } 
    }
}

int main()
{
    cout<<"Generating prime numbers. Please Wait...";
    SieveOfEratosthenes();
    int testCases;
    cout<<"\nEnter number of test cases: ";
    cin>>testCases;
    while (testCases--)
    {
        cout<<"\nEnter number of terms of Fibonacci Series less than 40: ";
        long long n;
        cin>>n;
        if(n<1 || n>=40)
        {
            cout<<"\n Invalid input!!";
            continue;
        }
        cout<<"Twisted Fibonacci series: 1 ";
        if(n==1)
        {
            continue;
        }
        cout<<"1 ";
        if(n==2)
        {
            continue;
        }
        vector<long long> fibSeries;
        fibSeries.push_back(1);
        fibSeries.push_back(1);
        long long ctr=2;
        while (ctr<n)
        {
            long long res = fibSeries[ctr-1]+fibSeries[ctr-2];
            if(PrimeNumbers[res])
            {
                cout<<"0 ";
            }
            else if(res%5==0)
            {
                cout<<"0 ";
            }
            else
            {
                cout<<res<<" ";
            }
            fibSeries.push_back(res);
            ctr++;
        }
    }
    return 0;
}

File: Python/6thOct-2.py
def fibonacci(n):
	l=[1]
	a = 0
	b = 1
	if n <= 0:
		print("Incorrect input")
	elif n == 1:
		print(b) 
	else:
		for i in range(1,n):
			c = a + b
			a = b
			b = c
			if b%5==0:
				l.append(0)
			else:
				l.append(b)
		isPrime(l) 



def isPrime(l):  
	l1=[]
	for n in l:
		flag=0
		if n <= 1:
			pass
		for i in range(2,n):
			if n % i == 0:
				flag=1
				break
		if flag==0 and n>2:
			l1.append(0)
		else:
			l1.append(n)
	print(l1)



	
fibonacci(12)
print()
fibonacci(27)	
	
	

